### PR TITLE
media: Return 0 when OutputDatasource size is 0

### PR DESCRIPTION
**Pull request by junmin-kim**
0 length fwrite(FileOutputDataSource) or array(BufferOutputSource)
are not verified. So just return 0 if size is 0

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [38439d823913aa9c0fe4b4645f68e61a99b48721] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [38439d823913aa9c0fe4b4645f68e61a99b48721] - Code Rule Check (C++) OK.

File: framework/src/media/FileOutputDataSource.cpp [hunk 1]
```diff
@@ -113,6 +113,10 @@ ssize_t FileOutputDataSource::write(unsigned char *buf, size_t size)
 		return EOF;
 	}
 
+	if (size == 0) {
```
Comment (davidfather) [inline framework/src/media/FileOutputDataSource.cpp]: Why don't you add this fix to SocketOutputDataSource as well?

File: framework/src/media/BufferOutputDataSource.cpp [hunk 2]
```diff
@@ -86,6 +86,10 @@ ssize_t BufferOutputDataSource::write(unsigned char *buf, size_t size)
 		return EOF;
 	}
 
+	if (size == 0) {
```
Comment (Taejun-Kwon) [inline framework/src/media/BufferOutputDataSource.cpp]: Move this  state to top of write function.
Cheapest first, It is good habit.

**Comment by seinfra** [comment]
Target : [c7ea659ad6f99df300b3e3ebc0967d458f55b1bd] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [c7ea659ad6f99df300b3e3ebc0967d458f55b1bd] - Code Rule Check (C++) OK.

**Pull request merged**